tools/add_triage_signature.py

```diff
@@ -325,6 +333,78 @@ def _update_ticket(self, url, issue_key, should_update=False):
         self._update_description(issue_key, description)
 
 
+class FailureDetails(Signature):
+    def __init__(self, jira_client):
+        super().__init__(jira_client, comment_identifying_string="")
+
+    def is_olm_operator(self, operator_name):
+        return operator_name.lower() in ["cnv", "lso", "ocs"]
+
+    def _update_ticket(self, url, issue_key, should_update=False):
+
+        #import ipdb; ipdb.set_trace()
+        def extract_cluster_md_from_existing_labels():
+            label_prefix_to_property = {
+                "AI_CLUSTER_": "id",
+                "AI_DOMAIN_": "email_domain",
+                "AI_USER_": "user_name",
+            }
+            r = re.compile(r'(AI_[^_]*_)(.*)')
+
+            i = self._jclient.issue(issue_key)
+            cluster_md = {}
+            for l in i.fields.labels:
+                m = r.match(l)
+                if m is None or len(m.groups()) != 2:
+                    continue
+                prefix, value = m.groups()
+                if prefix not in label_prefix_to_property:
+                    continue
+                cluster_md[label_prefix_to_property[prefix]] = value
+            return cluster_md
+
+        if not should_update:
+            logger.debug("Not updating fields of %s", issue_key)
+            return
+
+        url = self._logs_url_to_api(url)
+        cluster_md = []
+        try:
+            md = get_metadata_json(url)
+            cluster_md = md['cluster']
+        except:
+            # if we cannot find the failure logs on the log-server, we'll just use whatever information we
+            # have in the 'labels' field of the ticket
+            # this is mainly relevant for triage tickets that were moved from 'MGMT' project to 'AITRIAGE'
+            # project
+            cluster_md = extract_cluster_md_from_existing_labels()
+
+        update_fields = {
+            "labels": [],
+        }
+        if 'user_name' in cluster_md:
+            update_fields[CF_USER] = cluster_md['user_name']
+        if 'email_domain' in cluster_md:
+            cluster_md[CF_DOMAIN] = cluster_md['email_domain']
+        if 'id' in cluster_md:
+            cluster_md[CF_CLUSTER_ID] = cluster_md['id']
+
+        feature_usage_field = cluster_md.get('feature_usage')
+        if feature_usage_field:
+            feature_usage = json.loads(feature_usage_field)
+            operators = [feature for feature in feature_usage.keys() if self.is_olm_operator(feature)]
+            other_features = [feature for feature in feature_usage.keys() if not self.is_olm_operator(feature)]
+            if len(operators+other_features) > 0:
+                labels = []
+                for f in operators+other_features:
+                    labels.append("FEATURE-"+f.replace(" ", "-"))
```

Comment:
a bit of pep8

Reply:
pep8 does not seem to have an issue with this :)